CMakeLists.txt

```diff
@@ -197,6 +212,12 @@ set(_acts_nlohmanjson_version 3.2.0)
 set(_acts_onnxruntime_version 1.12.0)
 set(_acts_root_version 6.20)
 set(_acts_tbb_version 2020.1)
+#set(_acts_vecmem_version 1.4.0)
+#set(_acts_algebraplugins_version 0.22.0)
+#set(_acts_covfie_version 0.9.0)
+#set(_acts_detray_version 0.55.0)
+#set(_acts_traccc_version 0.8.0)
+#set(_acts_thrust_version 0.8.0)
```

Comment:
Did you hard-code these in the `<package>/CMakeFiles.txt`? I like that they are centrally set.

Reply:
I know I'm stirring a deeper pot right here, but I'm not the biggest fan of this setup. :frowning: I understand the appeal of seeing all version numbers side-by-side, but:
  - These "hidden variables" are passed around, and turned into actual cache variables in some really awkward ways at the moment. See for instance:

https://github.com/acts-project/acts/blob/main/thirdparty/actsvg/CMakeLists.txt

(That thing turns `_acts_actsvg_version` into `ACTS_VERSION`, and then the actual cache variable, `ACTS_ACTSVG_GIT_TAG`, which gets used by CMake.)

  - I switched both Athena and all the R&D projects to always download TGZ files instead of doing Git clones. With the ability for the user to switch to git clones if they absolutely want to. This is done by using a single `<FOO>_SOURCE` cache variable, like in:
    * https://github.com/acts-project/traccc/blob/main/extern/vecmem/CMakeLists.txt#L20-L22
    * https://gitlab.cern.ch/atlas/atlasexternals/-/blob/main/External/VecMem/CMakeLists.txt?ref_type=heads#L15-17
    * etc.

The latter, picking up TGZ files, is really hard to sync with specifying version number in a single place. :thinking: You'd have to specify the whole "source strings" in a single place (including the MD5 hashes when needed), or the whole thing just doesn't work.

So all in all, not the biggest fan of the current Acts setup. Reminds me way too much of what [lcgcmake](https://gitlab.cern.ch/sft/lcgcmake) does. (Which is not a good thing. :frowning:)